src/compiler/checker.ts

```diff
@@ -16367,7 +16367,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 error(node, diag, typeStr, minTypeArgumentCount, typeParameters.length);
                 if (!isJs) {
                     // TODO: Adopt same permissive behavior in TS as in JS to reduce follow-on editing experience failures (requires editing fillMissingTypeArguments)

```

Comment:
This is a pretty old TODO (see https://github.com/microsoft/TypeScript/pull/20232/files#r152707677 ). Based on a quick local test removing this branch altogether and just following through with the logic would return very similar results display-wise. But then all of those would became "valid" type references etc and not error types. 

I think this is something that requires input from the team. I think though that this TODO should either be removed or implemented in this PR - I'm just not sure what's the preference of the team.